framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2722,6 +2723,21 @@ private AnnotatedDeclaredType getFunctionalInterfaceType(Tree lambdaTree,
                 AnnotatedExecutableType methodExe = result.second;
                 return (AnnotatedDeclaredType) methodExe.getReturnType();
 
+            case CONDITIONAL_EXPRESSION:
+                ConditionalExpressionTree conditionalExpressionTree = (ConditionalExpressionTree) parentTree;
+                final AnnotatedTypeMirror falseType = getAnnotatedType(conditionalExpressionTree.getFalseExpression());
+                final AnnotatedTypeMirror trueType = getAnnotatedType(conditionalExpressionTree.getTrueExpression());
+                assertFunctionalInterface(javacTypes, (Type) falseType.getUnderlyingType(), parentTree, lambdaTree);
+                assertFunctionalInterface(javacTypes, (Type) trueType.getUnderlyingType(), parentTree, lambdaTree);
+                if (!falseType.equals(trueType)) {
```

Comment:
I think you're right.  Two things:
Should the right hand side really be the leastUpperBound of the true/false branches?  Secondly, I believe we will issue a typechecking warning later if they are not subtypes of the left-hand side so we don't actually have to report this?  
Finally using equals will compare the primary annotations which don't necessarily have to be equal.